File: Array/Easy/1365_How_Many_Numbers_Are_Smaller_Then_The_Current_Number/Solution.py
from typing import List


class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        maxElement = max(nums)
        vec = [0] * (maxElement + 1)
        for num in nums:
            vec[num] += 1

        for i in range(1, maxElement + 1):
            vec[i] += vec[i - 1]

        for i, num in enumerate(nums, 0):
            nums[i] = 0 if num == 0 else vec[num - 1]

        return nums

File: Array/Easy/1365_How_Many_Numbers_Are_Smaller_Then_The_Current_Number/Solution.cpp
#include <vector>
#include <algorithm>


class Solution {
public:
    std::vector<int> smallerNumbersThanCurrent(std::vector<int>& nums) {
        int max{*(std::max_element(nums.begin(), nums.end()))};
        std::vector<int> vec(max + 1);
        for (int i : nums) ++vec[i];
        
        for (int i{1}; i <= max; ++i) vec[i] += vec[i - 1];
        for (int i{}; i < nums.size(); ++i) {
            nums[i] = nums[i] == 0 ? 0 : vec[nums[i] - 1];
        }

        return nums;
    }
};